### ISSUE TITLE
mpris commands behave strange (react but time out / no reaction and time out)

### ISSUE THREAD

**Issue by timo** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#2321]
**OS:**

Fedora 25

**Issue Descriptions:**

I noticed this when I received a call and KDE Connect didn't pause my music. instead, after the call *ends* it pauses the music.

I used dbus-monitor to figure out what's going on and here's what i found:

KDEC asks for CanPause, which returns true, then calls Pause which does nothing.

When the call ends, KDEC uses PlayPause instead, which works, and causes the player to stop.

Trying to call the mpris methods manually via busctl (because i can't be arsed to figure out dbus-send) gives me "Connection timed out" for Pause, PlayPause, Next, Previous, Play, and Stop. Out of those commands the only one that does nothing is Pause.

**Steps to Reproduce:**

Call `busctl --user call org.mpris.MediaPlayer2.google-play-music-desktop-player /org/mpris/MediaPlayer2 org.mpris.MediaPlayer2.Player Pause`, notice that nothing happens, see that "Connection timed out" appears after a few seconds.

Call `busctl --user call org.mpris.MediaPlayer2.google-play-music-desktop-player /org/mpris/MediaPlayer2 org.mpris.MediaPlayer2.Player Stop` while music is running, notice that the music immediately stops, see that "Connection timed out" appears after a few seconds.

**Comment by timo** [comment]
```diff
diff --git a/src/main/features/linux/mprisService.js b/src/main/features/linux/mprisService.js
index dfc8e35..a555222 100644
--- a/src/main/features/linux/mprisService.js
+++ b/src/main/features/linux/mprisService.js
@@ -30,6 +30,12 @@ function mprisService() {
     }
   });
 
+  player.on('pause', () => {
+    if (PlaybackAPI.isPlaying()) {
+      Emitter.sendToGooglePlayMusic('playback:playPause');
+    }
+  });
+
   player.on('playpause', () => {
     Emitter.sendToGooglePlayMusic('playback:playPause');
   });
```

here's a patch that makes it react to "pause" by pausing the music.

**Comment by MarshallOfSound** [comment]
@timo PR would be great 😄

**Comment by jostrander** [comment]
Yeah that's totally an oversight on my part. I was only working with the task bar integrations for cinnamon and gnome which weren't using pause.

**Comment by MarshallOfSound** [comment]
Fixed in #2323 

Thanks @timo 👍
